### PR TITLE
FS-50: Gift codes

### PR DESCRIPTION
**Pull request by sugargoat**
### Motivation

Gift codes allow MobileCoin payments in the form of a gift that can be claimed by anyone who has the gift code.

### In this PR
* Adds gift code service and db models

[FS-50](https://mobilecoin.atlassian.net/browse/FS-50)

### CODE REVIEW

File: full-service/src/service/account.rs [hunk 1]
```diff
@@ -17,7 +17,57 @@ use mc_fog_report_validation::FogPubkeyResolver;
 use mc_ledger_db::Ledger;
 use mc_util_from_random::FromRandom;
 
+use crate::service::ledger::LedgerServiceError;
 use diesel::Connection;
+use displaydoc::Display;
+
+#[derive(Display, Debug)]
+pub enum AccountServiceError {
```
Comment (sugargoat) [inline full-service/src/service/account.rs:25]: Opportunistically isolated AccountServiceErrors - no change in functionality here.

File: API.md [hunk 2]
```diff
@@ -30,6 +30,11 @@ The Full Service Wallet API provides JSON RPC 2.0 endpoints for interacting with
 * [get_all_transaction_logs_ordered_by_block](#get-all-transaction-logs-ordered-by-block)
 * [get_proofs](#get-proofs)
 * [verify_proof](#verify-proof)
+* [build_gift_code](#build-and-submit-gift-code)
```
Comment (christian-oudard) [inline API.md:33]:
```suggestion
* [build_gift_code](#build-gift-code)
```
broken link

**Review by christian-oudard — APPROVED** [review]
Really cool to read about how gift codes don't require any help from consensus, just create a throwaway account and send the recipient the entropy.

**Comment by onny** [comment]
Hey I would like to test this feature :) How does it work?

**Pull request merged**